src/OceanSeaIceModels/InterfaceComputations/compute_interface_state.jl

```diff
@@ -17,11 +18,31 @@ end
     return !(converged | reached_maxiter) | hasnt_started
 end
 
+# Fixed number of iterations
 struct FixedIterations
     iterations :: Int
 end
 
-@inline iterating(Ψⁿ, Ψ⁻, iteration, fixed::FixedIterations) = iteration < fixed.iterations
+@inline iterating(Ψⁿ, Ψ⁻, ℙᵢ, iteration, fixed::FixedIterations) = iteration < fixed.iterations
+
+# Stop if the surface temperature is above freezing
+struct SeaIceConvergenceStopCriteria{FT}
+    tolerance :: FT
+    maxiter :: Int
+end
+
+@inline function iterating(Ψⁿ, Ψ⁻, ℙᵢ, iteration, convergence::SeaIceConvergenceStopCriteria)
+    maxiter = convergence.maxiter
+    tolerance = convergence.tolerance
+    hasnt_started = iteration == 0
+    reached_maxiter = iteration ≥ maxiter
+    drift = abs(Ψⁿ.u★ - Ψ⁻.u★) + abs(Ψⁿ.θ★ - Ψ⁻.θ★) + abs(Ψⁿ.q★ - Ψ⁻.q★)
+    Tₘ = ℙᵢ.liquidus.freshwater_melting_temperature
+    Tₘ = convert_to_kelvin(ℙᵢ.temperature_units, Tₘ)
+    above_melting = Ψⁿ.T ≥ Tₘ
+    converged = drift < tolerance
+    return !(converged | reached_maxiter | above_melting) | hasnt_started 
```

Comment:
Does this make sense? The intermediate state shouldn't be assumed to be physically meaningful. The temperature may be intermediate above melting but not converged?

This could cause problems with initialziation too because you can't escape from an above freezing initial condition.

Reply:
I also think this would prevent a situation where the surface melting temperature is a function of some state variable, like salinity. Changes in salinity, which change the melting temperature, would cause the solver to fail.

Reply:
It's also not very logical. "Temperature above melting" has nothing to do with "convergence". I think that expresses the basic flaw in this

Reply:
Without this change the solver will not converge because melting temperature is not in equilibrium with the fluxes. The non equilibrium part of the flux is used to melt sea ice which would not melt otherwise.

Reply:
> It's also not very logical. "Temperature above melting" has nothing to do with "convergence". I think that expresses the basic flaw in this

I don't understand what you mean by this, I don't think it is relevant with this change. The solver is trying to find a temperature which is in equilibrium with incoming fluxes. In certain situations this temperature is _above_ melting. This condition cannot be found by the solver if we continue manually capping the temperature. In this case (since we are in a non-equilibrium condition) we need to stop the solver.